tests/calculation/test_partial_charge.py

```diff
@@ -323,6 +325,32 @@ def test_stm_default_settings(PolarizedNonSplitPartialCharge):
     assert actual == defaults
 
 
+def test_stm_updated_settings(PolarizedNonSplitPartialCharge):
+    defaults = {
+        "sigma_xy": 4.0,
+        "sigma_z": 4.0,
+        "truncate": 3.0,
+        "enhancement_factor": 1000,
+        "interpolation_factor": 10,
+    }
+    new_settings = {
+        "sigma_xy": 3.0,
+        "sigma_z": 5.0,
+        "truncate": 2.0,
+        "enhancement_factor": 2000,
+        "interpolation_factor": 20,
+    }
+    settings = PolarizedNonSplitPartialCharge.stm_settings
+    print(type(settings))
+    default_settings = dataclasses.asdict(settings)
+    assert default_settings == defaults
+    settings = STM_settings(**new_settings)
+    print(settings)
+    PolarizedNonSplitPartialCharge.to_stm(stm_settings=settings)
+    updated_settings = dataclasses.asdict(PolarizedNonSplitPartialCharge.stm_settings)
+    assert updated_settings == new_settings
```

Comment:
Instead test whether passing the settings to the `to_stm` method modifies the resulting plot.